Test/ElasticScale.ShardManagement.UnitTests/ScenarioTests.cs

```diff
@@ -1237,6 +1237,119 @@ public void BasicScenarioRangeShardMaps()
+                string shardMapName = "PerTenantShardMap";
+
+                #region Setup
+
+                // Deploy shard map manager.
+                ShardMapManager shardMapManager = ShardMapManagerFactory.CreateSqlShardMapManager(
+                    Globals.ShardMapManagerConnectionString,
+                    ShardMapManagerCreateMode.ReplaceExisting);
+
+                // Create a single user per-tenant shard map.
+                ListShardMap<int> perTenantShardMap = shardMapManager.CreateListShardMap<int>(shardMapName);
+
+                ShardLocation sl1 = new ShardLocation(
+                        Globals.ShardMapManagerTestsDatasourceName,
+                        ScenarioTests.s_perTenantDBs[0]);
+
+                // Create first shard and add 1 point mapping.
+                Shard s = perTenantShardMap.CreateShard(sl1);
+
+                // Create the mapping.
+                PointMapping<int> p1 = perTenantShardMap.CreatePointMapping(1, s);
+
+                #endregion Setup
+
+                // Delete and recreate perf counter catagory.
+                ShardMapManagerFactory.CreatePerformanceCategoryAndCounters();
+
+                // Eager loading of shard map manager
+                ShardMapManager smm = ShardMapManagerFactory.GetSqlShardMapManager(Globals.ShardMapManagerConnectionString, ShardMapManagerLoadPolicy.Eager);
+
+                // check if perf counter instance exists, instance name logic is from PerfCounterInstance.cs
+                string instanceName = string.Concat(Process.GetCurrentProcess().Id.ToString(), "-", shardMapName);
+
+                Assert.IsTrue(ValidateInstance(instanceName));
+
+                // verify # of mappings.
+                Assert.IsTrue(ValidateCounterValue(instanceName, PerformanceCounterName.MappingsCount, 1));
+
+                ListShardMap<int> lsm = smm.GetListShardMap<int>(shardMapName);
+
+                // Add a new shard and mapping and verify updated counters
+                ShardLocation sl2 = new ShardLocation(
+                        Globals.ShardMapManagerTestsDatasourceName,
+                        ScenarioTests.s_perTenantDBs[1]);
+
+                Shard s2 = lsm.CreateShard(sl2);
+
+                PointMapping<int> p2 = lsm.CreatePointMapping(2, s2);
+                Assert.IsTrue(ValidateCounterValue(instanceName, PerformanceCounterName.MappingsCount, 2));
+
+                // perform DDR operation few times and validate non-zero counter values
+                for (int i = 0; i < 10; i++)
+                {
+                    using (SqlConnection conn = lsm.OpenConnectionForKey(1, Globals.ShardUserConnectionString))
+                    {
+                    }
+                }
+
+                Assert.IsTrue(ValidateNonZeroCounterValue(instanceName, PerformanceCounterName.DdrOperationsPerSec));
+                Assert.IsTrue(ValidateNonZeroCounterValue(instanceName, PerformanceCounterName.MappingsLookupSucceededPerSec));
```

Comment:
I like this testing. Can you add basic sanity-level checks like these for the other perf counters?